### PR TITLE
Ia/feature/pdf export

### PR DESCRIPTION
**Pull request by andhdo**
Simple export-to-pdf functionality using clj-pdf

###### Before submitting the PR, please make sure you do the following 
-  [x] If there are changes to the backend codebase, run the backend tests with `lein test && lein eastwood && lein bikeshed && lein docstring-checker && ./bin/reflection-linter`
-  [x] Run the frontend and integration tests with  `yarn && yarn run prettier && yarn run lint && yarn run flow && ./bin/build version uberjar && yarn run test`)
-  [x] Sign the [Contributor License Agreement](https://docs.google.com/a/metabase.com/forms/d/1oV38o7b9ONFSwuzwmERRMi9SYrhYeOrkbmNaq9pOJ_E/viewform)
(unless it's a tiny documentation change).

### CODE REVIEW

**Comment by salsakran** [comment]
@andhdo Thank you for hour help!

The pull request template includes a few things we need from you, specifically making sure that tests pass, that linting passes and that you have signed our CLA.

Also, this PR is missing tests. Please add a few so future code changes don't break this.

**Review by salsakran — CHANGES_REQUESTED** [review]
Thanks @andhdo for the PR.

However, for us to accept this PR we need you to 
1. Sign the CLA
2. Add tests for this functionality
3. Verify that this doesn't break any other functionality by running the full test suite. 

We include these instructions in the PR template, and can not accept any PR that doesn't include the above steps.

**Comment by andhdo** [comment]
Added modifications for testing & lint purposes.

**Comment by tlrobinson** [comment]
@salsakran @kdoh @mazameli Currently the PDF output is essentially unstyled. Do we want to give it a little Metabase design love? I'm not sure how difficult that will be, as I haven't looked closely at [clj-pdf](https://github.com/clj-pdf/clj-pdf)

e.x.

screen shot 2018-04-02 at 11 10 14 am

**Comment by tlrobinson** [comment]
The download popover is also getting a bit wide:

screen shot 2018-04-02 at 11 26 53 am

**Comment by mazameli** [comment]
Looks like this is how you do styling: https://github.com/clj-pdf/clj-pdf#stylesheets Would it make more sense to merge this and do the styling after in a follow-up PR, or try to do styling in this PR?

And yeah, for the download menu we probably want to stack them vertically for now. We'll be moving the download actions when we implement [entity menus](https://github.com/metabase/metabase/issues/6217) in entity pages.

**Comment by salsakran** [comment]
Strong 👎 on an unstyled pdf download.

What's the underlying use case for this PR @andhdo ? Are you using this yourself internally?

File: test/metabase/api/public_test.clj
Comment (salsakran) [inline test/metabase/api/public_test.clj]: Is there a way to check that the pdf contains what you think it contains?

**Comment by andhdo** [comment]
Hi; some answers. I apologize the inconveniences. 
1. @tlrobinson - about the popover: it was a way to show the option, because with a fixed size, the link cannot be seen.
2. @tlrobinson, @salsakran  - about unstyled: it was the most basic way that i'd found to generate the output.
3. @salsakran - about internal usage: it is used to save some evidence or snapshot about the result.
4. @salsakran - about the test: unfortulately i couldn't found an example inside the clj-pdf test to verify the presence of a table inside the generated report.

I'll gonna try to modify the 2nd and 5th points. I hope to collaborate with the few hours left for this and learn a little bit of clojure, language that I did not previously know.

**Comment by andhdo** [comment]
@salsakran ; i made some changes according the assestments made by you and coworkers; Please, let me know if anything else is required.

**Comment by BaelishLord** [comment]
Is this feature now available?
how can i run metabase to get this feature?

**Comment by andhdo** [comment]
@BaelishLord: you can do a manual merge of the  [ia/feature/pdf-export](https://github.com/in-action/metabase/tree/ia/feature/pdf-export) branch with the metabase master, then compile and run it.

**Comment by kdoh** [comment]
Sorry for the long wait on this @andhdo. As soon as we start getting into any sort of rich export option like a PDF, the user expectation will most likely be that formatting settings present in the Metabase UI for those results would apply to the exported PDF. Give all that, I don't think we'll be able to take this PR in its current form and given the complexities involved I think it is probably something the core team should tackle unless you're willing to sign up for a lot of back and forth and design review. I'm not sure that a PDF of a table without that, even with minimal styling is enough of a lift to be worth the additional code or cruft to the user experience and so I'm going to close this. Thanks again for looking into this and raising the issue.

**Pull request closed without merging**
